File: app/src/main/java/com/lumiyaviewer/lumiya/render/GLTexture.java
package com.lumiyaviewer.lumiya.render;

import android.graphics.Bitmap;
import java.nio.ByteBuffer;

public interface GLTexture {
    int SetAsTexture();

    Bitmap getAsBitmap();

    byte getByte(int i);

    ByteBuffer getExtraComponentsBuffer();

    int getHeight();

    int getNumComponents();

    int getRGB(int i);

    int getWidth();
}

File: app/src/main/java/com/lumiyaviewer/lumiya/render/GLTexture.kt
package com.lumiyaviewer.lumiya.render

import android.graphics.Bitmap
import java.nio.ByteBuffer

interface GLTexture {
    @Suppress("FunctionName") fun SetAsTexture(): Int
    fun getAsBitmap(): Bitmap
    fun getByte(index: Int): Byte
    fun getExtraComponentsBuffer(): ByteBuffer
    fun getHeight(): Int
    fun getNumComponents(): Int
    fun getRGB(index: Int): Int
    fun getWidth(): Int
}
